gnucash/gnome/dialog-print-check.c

```diff
@@ -2112,7 +2117,39 @@ draw_page_items(GtkPrintContext *context,
     trans = xaccSplitGetParent(pcd->split);
     /* This was valid when the check printing dialog was instantiated. */
     g_return_if_fail(trans);
-    amount = gnc_numeric_abs(xaccSplitGetAmount(pcd->split));
+
+    if (pcd->account)
+    {
+        /* A parent account, e.g. from a subaccount register plugin page.
+         * Subtotal the amount of all splits from children accounts. */
+        GList *children_accounts = NULL;
+        GList *node, *child;
+        children_accounts = gnc_account_get_descendants(pcd->account);
+        children_accounts = g_list_append(children_accounts, pcd->account);
+        amount = gnc_numeric_zero();
+
+        for (node = xaccTransGetSplitList(trans); node; node = node->next)
+        {
+            Split *split = node->data;
+            Account* acct = xaccSplitGetAccount(split);
+            for (child = children_accounts; child; child = child->next)
+            {
+                if (acct == child->data)
+                {
+                    amount = gnc_numeric_add_fixed(amount, xaccSplitGetAmount(split));
+                    break;
+                }
+            }
```

Comment:
Don't loop on all of pcd->account's children, use

    if (pcd->account == xaccSplitGetAccount(split) ||
        pcd->account == gnc_account_get_parent(xaccSplitGetAccount(split)))
        amount = gnc_numeric_add_fixed(amount, xaccSplitGetAmount(split));

Reply:
I have like 5 levels of subaccounts... won't `gnc_account_get_parent` return only the direct parent?

Reply:
Not pointing fingers, but I copied this code from the existing ledger display logic:
https://github.com/Gnucash/gnucash/blob/4da655d0f996d486a187360b4e0a5b64ee3eaa69/gnucash/register/ledger-core/split-register-model.c#L112-L126

Reply:
So? There's lots of crappy code in GnuCash. That's no excuse for adding more.

Reply:
> I have like 5 levels of subaccounts... won't `gnc_account_get_parent` return only the direct parent?

Yup. But `gnc_account_get_children` also goes only one level. You'd need to use `gnc_account_get_descendants` or recurse on `gnc_account_get_children`, which is what `get_descendants` does under the hood. Either gets ugly so you might consider `gnc_account_foreach_descendant` with a test-and-accumulate function that checks each account in turn for splits in the transaction rather than the other way around. That way you're doing the more expensive loop only once instead of once per split.

Reply:
Sorry, not being defensive! I'm happy to use whatever best practices you'd prefer. I had simply found some existing code that did what I needed and copied it. However, both do use `gnc_account_get_descendants` to get all children accounts recursively into a list...

> you might consider `gnc_account_foreach_descendant` with a test-and-accumulate function that checks each account in turn for splits in the transaction rather than the other way around. That way you're doing the more expensive loop only once instead of once per split.

Done. Found a few other helpers so all manual loops are removed.